### PR TITLE
Support LV2 with MSVC

### PR DESCRIPTION
**Pull request by DomClark**
Allows building LV2 support for Windows using MSVC. Adds the required vcpkg ports to `.appveyor.yml`, changes `CMakeLists.txt` to be able to use these ports, and makes a few small changes required to successfully compile under MSVC.

### CODE REVIEW

**Comment by superpaik** [comment]
I don't know if that is correct for testing, but I installed the Windows 64-bit made by the boot and lv2 pluggins are not available. On the LMMS starting page you can read "Lv2 plugin SUMMARY: 0 of 0  loaded in 0 msecs"
imatge

**Comment by DomClark** [comment]
LMMS doesn't ship with any LV2 plugins by default. You can install some free ones for Windows from https://x42-plugins.com/x42/, for example.

**Comment by superpaik** [comment]
Sorry, my fault.
I installed a couple of effects (using "Windows 64-bit: lmms-1.2.2-msvc2017-win64.exe" for testing) and it worked ok.

**Review by JohannesLorenz — APPROVED** [review]
I left only one minor comment about the `STATUS_LV2` message in CMake. LGTM.

File: CMakeLists.txt
Comment (DomClark) [reply CMakeLists.txt:216]: I didn't really think about this - I just moved the message down from above. I would argue that "set `PKG_CONFIG_PATH` appropriately" implies that pkg-config needs to be installed - it can't be set appropriately if there's no pkg-config to point it at. I'm not completely familiar with how everything is acquired and found on other platforms, so I don't know what would be better to put here, but I'm open to suggestions.

Comment (JohannesLorenz) [reply CMakeLists.txt:216]: The "set `PKG_CONFIG_PATH` appropriately" argument makes sense. About the `FIND_PACKAGE(LV2 CONFIG)`, I don't know what else could lead to failure here (probably it's covered by "install it"?).

I think this comment is not a blocker, we can still change the message if users complain about not understanding the error.

Comment (DomClark) [reply CMakeLists.txt:216]: If the CMake config file isn't found, and vcpkg is being used, then either:
* The relevant ports aren't installed; this is already mentioned in the message,
* The vcpkg toolchain file isn't specified correctly, in which case there will be a bunch of other errors too, or
* There's a bug in vcpkg or in one of the ports, in which case they should come and speak to us anyway.

I'm going to resolve this conversation; feel free to unresolve it if there's something else that needs to be brought up.

**Comment by DomClark** [comment]
Will merge in a day or two if there are no objections.

File: CMakeLists.txt
Comment (PhysSong) [inline CMakeLists.txt:205]: Just wanted to mention an unusual scenario: what if `lv2` installation provides pkg-config file only whereas `lilv` provides CMake config file only? I know that won't happen for normal users, so please feel free to ignore.

File: CMakeLists.txt
Comment (PhysSong) [inline CMakeLists.txt:209]: You don't have to check for `LV2_FOUND` here, right?

File: CMakeLists.txt
Comment (DomClark) [reply CMakeLists.txt:205]: I considered that, but didn't think it was worth handling. Also, we probably want to fail in that situation anyway. `lilv` should depend on `lv2`, so there would be a risk of having headers for two different versions of `lv2` in the include path.

File: CMakeLists.txt
Comment (DomClark) [reply CMakeLists.txt:209]: Good point, will fix. 👍

File: CMakeLists.txt
Comment (PhysSong) [reply CMakeLists.txt:205]: > `lilv` should depend on `lv2`, so there would be a risk of having headers for two different versions of `lv2` in the include path.

Makes sense.

File: CMakeLists.txt
Comment (DomClark) [reply CMakeLists.txt]: Fixed, and rebased on master while I was at it.

File: CMakeLists.txt
Comment (DomClark) [reply CMakeLists.txt:216]: This PR doesn't change the error message - this one I moved down from earlier in the file, and the "requires pkg-config" one I removed since it's no longer true. I'll leave this up to @JohannesLorenz, in the case that we do want to fix it here.

**Comment by JohannesLorenz** [comment]
@DomClark for your info (it's not a comment stopping this PR): We have a new [Lv2 wiki page](https://github.com/LMMS/lmms/wiki/lv2). If compiling LMMS with Lv2 (or installing Lv2) is nontrivial, you can add the steps there. Also, you can add instructions how to install the plugins on Windows, if it's not obvious.

File: CMakeLists.txt
Comment (tresf) [reply CMakeLists.txt:205]: Thanks, I checked Ubuntu 18.04 as well as Homebrew and can confirm that both ship pkg-config files, neither ship cmake files. Thanks for the explanation.

File: CMakeLists.txt
Comment (JohannesLorenz) [reply CMakeLists.txt:216]: It's correct, the PR diff just looks like the error message was changed. It's from #4899 . Though, technically, we could change it here.

Imagine the user has the Lv2 pkg-config path installed outside of the systems pkg-config paths (local install) and gets "please install Lv2". I wonder if that makes clear that the message means "Yes, Lv2 may be installed, but I did not find it, give me the path where I need to search". What do you think, is it obvious by saying "please install Lv2"?

**Comment by tresf** [comment]
> Merge?

Yes.  I pointed out some wording that seems to be geared towards systems without `pkg-config`, which I believe @JohannesLorenz originally authored when `suil` was added in https://github.com/LMMS/lmms/commit/2a66e83f536ed71984054392ed469991de67874e, but shouldn't hold this up.  I'm just a bit confused who this wording was intended to target since `pkg-config` has been a requirement of LMMS as long as I can remember. :)

File: CMakeLists.txt
Comment (tresf) [reply CMakeLists.txt:216]: @JohannesLorenz oh, so the fear is that a particular system may not have it provided and `PKG_CONFIG_PATH` needs to be added?  That makes sense, especially for packages that may not be available by the package managers.  I tested on 18.04 and Homebrew, both which provide this file.  16.04's package listing seems to offer it as well.  I think `PKG_CONFIG_PATH` is more of an edge-case reminder for things that the package manager doesn't provide (e.g. Carla) but even the edge-cases we've managed to workaround without custom PKG_CONFIG_PATHs.  I think it's safe to remove, or perhaps [added to the build summary notes](https://github.com/LMMS/lmms/blob/9d104b2205f63bd4425905e1261ffd63f57baba9/CMakeLists.txt#L761-L765).

**Comment by DomClark** [comment]
@JohannesLorenz Thanks for the wiki link - I'll add the relevant information once this is merged. Compiling with LV2 should be easy (just `vcpkg install lilv lv2`), but the ports are quite new so anyone who's compiled LMMS on Windows before may need to update vcpkg first. Installing LV2 plugins is easy enough - the hard part is finding them, since it's not a common format on Windows and there's no package manager to search for them. I'll add a link to x42's plugins to get people started.

File: CMakeLists.txt
Comment (JohannesLorenz) [reply CMakeLists.txt:216]: @tresf Agreed, it's a general thing that also affects other libs (like Carla), so I'm OK with removing it. @DomClark can you please change the error message?

Comment (tresf) [reply CMakeLists.txt:216]: I'm also fine reverting this and I can fix the wording in a separate commit since `suil` should also be fixed.

Comment (JohannesLorenz) [reply CMakeLists.txt:216]: Revert? It's not introduced in this PR.

Separate commit looks cleaner to me, too, as this is not part of supporting Lv2 on MSVC.

Comment (tresf) [reply CMakeLists.txt:216]: > Revert? It's not introduced in this PR.

Yes, sorry the diff keeps throwing me off.  @DomClark's stated that at least twice.  Marking as resolved. :)

**Pull request merged**
